### DISCUSSION TITLE
realpath when given file with ending '/'

### DISCUSSION THREAD

**Discussion by EvenOAndersen** [emscripten-core/emscripten#22335] [General]
When doing:
```
fclose(fopen("**/tmp/file.txt**", "wt"));

char *rpr = realpath("**/tmp/file.txt/**",  ....)
```

The result is "**/tmp/file.txt**"

Is this correct ?

**Comment by sbc100** [comment]
I tried running the above code and it failed because `fopen` fails (because `**/tmp` directory doesn't exist)..

However I modified that above example to use `/tmp` instead and it seemed to behave correctly:

```
$ cat test.c
#include <stdio.h>
#include <stdlib.h>
#include <limits.h>

int main() {
  FILE* f = fopen("/tmp/file.txt**", "wt");
  printf("%p\n", f);
  if (!f) return 0;
  fclose(f);
  char *rpr = realpath("/tmp/file.txt/**",  NULL);
  printf("rpr: %s\n", rpr);
  return 0;
}
$ emcc test.c
$ node ./a.out.js 
0x11278
rpr: (null)
```

Could you share full failing example perhaps?

**Reply by EvenOAndersen** [reply]
Sorry about that. The '*' was added when I made the filename bold.
The problem, is a filename with a single ending '/' (no stars) like this:


```
include <stdio.h>
#include <stdlib.h>
#include <limits.h>

int main() {
  FILE* f = fopen("/tmp/file.txt", "wt");
  printf("%p\n", f);
  if (!f) return 0;
  fclose(f);
  char *rpr = realpath("/tmp/file.txt/",  NULL);
  printf("rpr: %s\n", rpr);
  return 0;
}
```

**Reply by EvenOAndersen** [reply]
and the expected result is nullptr

**Reply by sbc100** [reply]
@hoodmane, this looks like the kind of issue you have been working on recently.   Do you know if you have any outstanding changes here that might fix this?

**Reply by hoodmane** [reply]
Right, I still reproduce it on tip of tree. I will see if I can fix it.

**Reply by sbc100** [reply]
Looks like I had a PR back in august: https://github.com/emscripten-core/emscripten/pull/22352.  Not sure if its worth keeping/resurrecting.

**Comment by Sainul007** [comment]
Ok, alright
